### ISSUE TITLE
lxml cannot be installed in LibreOffice Python on Windows

### ISSUE THREAD

**Issue by jkornelsen** [silnrsi/oxttools#8]
Is there a reason why the code must use `lxml` rather than a different XML library? This makes it difficult to use on Windows and nearly impossible from within LibreOffice on Windows. I hope to demonstrate that executing from within LibreOffice can make things easier by providing dialogs rather than a command-line interface.

If it works out, I will add a pull request that does not use `lxml`. Then the *LingTools add-on* could include dialogs to call `oxttools` commands. A prototype dialog already exists, only functionality is minimal due to the `lxml` problem.

When I suggested this before, you said that `oxttools` creates language-specific extensions, while *LingTools* works across languages. It sounded like you may have misunderstood how it would work. A *LingTools* dialog can call `oxttools` in order to make one or more language-specific extensions. The resulting extensions are completely separate from *LingTools*. They are identical to extensions created by `oxttools` from the command-line.

**Comment by mhosken** [comment]
Bad news. Oxttools needs lxml because the xpath support in xml.etree is
woefully inadequate. If you can find a library that has full xpath that
works for libo, then I'd be happy to switch.

Is this a 32/64 bit issue or python version thing?

**Comment by jkornelsen** [comment]
> Is this a 32/64 bit issue or python version thing?

No, the issue is that installing new modules is not practical because LibreOffice uses a separate Python on Windows. The only way would be a lot of hacking of paths and variables to switch LibreOffice to a downloaded Python, not something that even most power users would be up for.

On Ubuntu, it's not a problem because LibreOffice uses the system-wide Python, although installing `lxml` is still a little inconvenient.

My code in progress is at https://github.com/jkornelsen/oxttools/tree/without_lxml. So far I've switched it to use `xml.etree` and added some hacks to handle `CDATA` and `XPath` variables and function extensions. Hard to tell at this point whether that will be good enough or not. I'm still learning about all the ways that `oxttools` uses `Xpath` and what things are not supported by `xml.etree`.

**Comment by DavidLRowe** [comment]
@jkornelsen Not sure if you're still working on this or not. If you find an xml library or can code a work-around, you can submit a pull request, or raise an issue (or reopen this one). For the time being, I'll close this issue.

**Comment by jkornelsen** [comment]
Hi David, this is still a big issue for me. I coded a solution using the
standard xml.etree library, and the code is here:
https://github.com/silnrsi/libreoffice-linguistic-tools/tree/master/LinguisticTools/pythonpath/oxttools.
It extends xml.etree to add some functionality that is needed for the XML
files used by LO dictionaries.

However, Martin did not want to use it when I asked about a year ago, I
believe because he felt that it would not be able to handle certain XML
file changes that he is hoping to incorporate into oxttools at some point
in the future. So at this point I am thinking of releasing it as part of
LOLT but keeping it separate from the main oxttools repo.

The code in the LOLT repo may not compile currently because I merged it
with the Python 3 only oxttools changes a couple of weeks ago and have not
tested it yet. (I may need to add support for Python 2 back in because of
Apache OpenOffice, but that's a job for another day).

-Jim K
